### PR TITLE
Chore(nginx-rock): Update the rock base, websockets version pin.

### PR DESCRIPTION
**Pull request by alithethird**
Updated the `indico-nginx-rock` base, updated the `nginx-prometheus-exporter` from a snap to source. Pinned the `websockets` version.

### Rationale

`indico-nginx-rock` base should have been updated when `indico-rock` base was updated so fixing that now.
`nginx-prometheus-exporter` snap was using old `golang` and that had some security issues.
`websockets` needs to be pinned.

### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: .trivyignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# pebble
+CVE-2024-34156
```
Comment (yanksyoon) [inline .trivyignore:2]: It seems it isn't needed here anymore :)

File: tox.ini [hunk 2]
```diff
@@ -102,6 +102,7 @@ deps =
     requests
     types-PyYAML
     types-requests
+    websockets<14.0 # https://github.com/juju/python-libjuju/issues/1184
```
Comment (yanksyoon) [inline tox.ini:105]: Would it be possible to briefly summarize the issue here please?
```suggestion
    ; 2024/11/19 - there is an incompatibility issue with latest websockets lib release and pylib juju
    websockets<14.0 # https://github.com/juju/python-libjuju/issues/1184
```

**Review by yanksyoon — DISMISSED** [review]
LGTM, minor comments :)

File: .trivyignore (hunk 1 above)
Comment (alithethird) [reply .trivyignore:2]: Yeah but looks like I should add more in there :( . Indico rock has several

File: tox.ini [hunk 3]
```diff
@@ -102,6 +102,8 @@ deps =
     requests
     types-PyYAML
     types-requests
+    ; 2024/11/19 - there is an incompatibility issue with latest websockets lib release and pylib juju
+    websockets<14.0 # https://github.com/juju/python-libjuju/issues/1184
```
Comment (arturo-seijas) [inline tox.ini:106]: The issue shows as closed. Hasn't a fix been released yet?

Comment (alithethird) [reply tox.ini:106]: It is still failing in the other PR: https://github.com/canonical/indico-operator/actions/runs/11891522173/job/33175938863?pr=460

**Review by jdkandersson — APPROVED** [review]

**Review by weiiwang01 — APPROVED** [review]

**Pull request merged**